### PR TITLE
fix: [CDS-44175]: Artifact source template CRUD support

### PR DESCRIPTION
**Pull request by samarth-sharma31**
### Summary

Mocks - https://www.figma.com/proto/rVIYkD5bbpZlcG2fSU69vO/Templates?page-id=6523%3A318915&node-id=6536%3A363841&viewport=-1158%2C-309%2C0.49&scaling=scale-down&starting-point-node-id=6536%3A363841&show-proto-sidebar=1

#### Screenshots

NA

Screenshot 2022-10-05 at 5 40 25 PM
Screenshot 2022-10-05 at 5 40 41 PM

Screenshot 2022-10-19 at 4 22 28 AM
Screenshot 2022-10-19 at 4 19 10 AM

Screenshot 2022-10-21 at 12 21 38 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 18 Issues 
* Code issue 18 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11693)

**Review by yc147 — DISMISSED** [review]

**Review by bhavya-sinha — DISMISSED** [review]

File: src/modules/72-templates-library/components/TemplateStudio/TemplateStudio.module.scss [hunk 1]
```diff
@@ -18,6 +18,8 @@
   height: calc(var(--page-min-height) - var(--page-header-height));
   padding: 0 !important;
   position: relative;
+  overflow-y: auto !important;
```
Comment (yc147) [inline src/modules/72-templates-library/components/TemplateStudio/TemplateStudio.module.scss:21]: why is this required?

Comment (samarth-sharma31) [reply src/modules/72-templates-library/components/TemplateStudio/TemplateStudio.module.scss]: This is not required anymore - have removed this change

**Review by yc147 — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]
